File: fixtures/demo-parent/src/bus.rs
pub struct Bus;

impl Bus {
    pub fn publish(&self) {
        let _ = crate::bus::events;
        encode();
        flush();
    }
}

pub fn encode() {
    flush();
}

pub fn flush() {}

/// Channel sink endpoint (plugin maps Channel-typed consts onto Endpoint).
pub const events: BroadcastChannel = BroadcastChannel;

pub struct BroadcastChannel;

File: fixtures/ts/pkg/bus.ts
class BroadcastChannel {}

const events: BroadcastChannel = new BroadcastChannel();

class Bus {
  publish(): BroadcastChannel {
    encode();
    flush();
    return events;
  }
}

function encode(): void {
  flush();
}

function flush(): void {}

export { Bus, events };
